scripts/utils/calculate-ARI.R

```diff
@@ -92,3 +92,100 @@ calculate_ari <- function(integrated_sce,
   return(ari_tibble)
 
 }
+
+#' Calculate Reverse ARI to compare clustering pre and post integration
+#'
+#' @param individual_sce_list List of SCE objects containing all batches to use for 
+#'   calculating reverse ARI. Must be the original, pre-merged object.
+#'   The list must be a named list with the batch names.
+#' @param integrated_sce The integrated SCE object
+#' @param seed Seed for initializing random sampling and clustering
+#' @param integration_method The name of the method that was used for integration
+#'    to create `integrated_sce`. One of: fastMNN, harmony, rpca, cca, scvi, or scanorama
+#' @param unintegrated Indicates whether the provided data is integrated (`FALSE`; default) or
+#'   integrated (`TRUE`).
+#' @param batch_column The variable in `integrated_sce` indicating the grouping of interest.
+#'  Generally this is either batches or cell types. Default is "batch".
+#'
+#' @return Tibble with three columns: `ari`, the calculated ARI, 
+#'   the corresponding `batch_id` found in the `batch_column` 
+#'   for the original SCE object, and the `integration_method`, the provided integration method
+#'
+calculate_reverse_ari <- function(individual_sce_list,
+                                  integrated_sce,
+                                  seed = NULL,
+                                  integration_method = NULL,
+                                  unintegrated = FALSE,
+                                  batch_column = "batch"){
+  set.seed(seed)
+  
+  # check that list of SCE objects is named
+  batch_ids <- names(individual_sce_list)
+  if(is.null(batch_ids)){
+    stop("Must provide a named list of SCE objects.")
+  } else {
+    # make sure the batch ids provided match between the list and the integrated object 
+    if(!(identical(sort(batch_ids), sort(unique(colData(integrated_sce)[, batch_column]))))){
```

Comment:
Can we space this out a bit just for legibility as follows? I'm sure if I suggest it formally here I'll break the spacing in a worse way, RStudio automatic tabbing to the rescue!
```
if(!(identical(
  ...first one...,
  ...second one...
)
```